### PR TITLE
working resnet18 in pspnet model

### PR DESCRIPTION
**Pull request by ibadami**
I have further fixed some bugs(which were causing runtime errors in resnet18) as follows:

- removed inline operations like d4+=e3 -> d4= d4+e3
- corrected kernel size, strides and padding for consistent input and output channels.
- fixed some minor typos like variable names etc
- commented line 16 in pspnet.py.  To be honest I was not sure what it does. If you need it you can uncomment it.

Upcoming(next 2 days):
- multiple resnet support: instead of hardcoded resnet18, different resnets can be used. In PSPnet two resnets are used as baseline, resnet 50 and resnet 101 and deeper resnets were inspected for performance improvement.
- optional dialation feature to switch it on and off with the choice of your resnet.
-pyramid pooling module
- auxiliary loss

### CODE REVIEW

**Comment by ibadami** [comment]
Umm, I am confuse to see the old commits showing up in this pull request.
I merged the master branch after the last pull was accepted. 
And while creating the new pull request I compared the two branches and only recent changes were shown.
Did I make some mistake? Let me know if this pull request is a repetition. I will create new pull request with the latest changes.

**Comment by meetshah1995** [comment]
I rebased and merged so that the changes reflect in your repo as well. Did you sync your fork with upstream after the PR was merged?

**Comment by meetshah1995** [comment]
The changes look good. I'll merge them once you sync the fork and update the PR.

Your upcoming contributions look good as well. 

It would however be great to directly use pre-trained torchvision ResNet layers [and inject the dilation into required Conv2D modules] rather than building the layers and then transferring the weights.  

Here's how the original implementation did it in Torch. [link](https://github.com/e-lab/LinkNet/blob/master/models/model.lua)

**Comment by meetshah1995** [comment]
There's now a `resnet-encoders` branch.

**Comment by ibadami** [comment]
Hi sorry for messy PR. I am closing this PR as I have incorporated this commit and more with a newer and cleaner PR. Thanks for being patient.
